include/ozo/ext/std/time_point.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include <ozo/pg/definitions.h>
+#include <ozo/io/send.h>
+#include <ozo/io/recv.h>
+
+#include <chrono>
+#include <iostream>
+
+namespace ozo::pg {
+
+// pg epoch is 2000-01-01 00:00:00, which is 30 years (10957 days)
+// after the unix epoch (the default value for a time_point)
+static constexpr const auto epoch = std::chrono::system_clock::time_point{} + std::chrono::hours{ 24 } * 10957;
```

Comment:
It is better to define `ozo::days`, `ozo::weeks`, `ozo::months`, `ozo::years` in and then define the epoch via this definitions (which would be using references for C++20 standard library).

Reply:
Yes, this would indeed be the nicest way to do it. This is, however, tricky to do with `std::chrono` in c++ < 20. It's fairly easy to define the concept of a day, since we have `std::chrono::hours` and a day is 24 hours. When you get to months it's more tricky, since the number of days in a month is not fixed.

The current implementation of a duration in `std::chrono` is that it's represented in a number of ticks, e.g. the number of microseconds. This is incompatible with the concept of months or years, since you can't represent them in a fixed number of these ticks.

Doing this would basically mean I can't use `std::chrono` (at least not in c++17).

Reply:
> Yes, this would indeed be the nicest way to do it. This is, however, tricky to do with std::chrono in c++ < 20. It's fairly easy to define the concept of a day, since we have std::chrono::hours and a day is 24 hours. When you get to months it's more tricky, since the number of days in a month is not fixed.

Yes, you right, this is a big pain even in ProstgreSQL time intervals :)